dataprofiler/profilers/profile_builder.py

```diff
@@ -444,6 +445,477 @@ def clean_data_and_get_base_stats(df_series, sample_size,
+
+            try:
+
+                data_labeler = DataLabeler(
+                    labeler_type='unstructured',
+                    dirpath=data_labeler_options.data_labeler_dirpath,
+                    load_options=None)
+                self.options.set(
+                    {'data_labeler.data_labeler_object': data_labeler})
+
+            except Exception as e:
+                utils.warn_on_profile('data_labeler', e)
+                self.options.set({'data_labeler.is_enabled': False})
+
+        if data is not None:
+            self.update_profile(data)
+
+    def __add__(self, other):
+        """
+        Merges two profiles together overriding the `+` operator.
+
+        :param other: unstructured profile being add to this one.
+        :type other: UnstructuredProfiler
+        :return: merger of the two profiles
+        """
+        if type(other) is not type(self):
+            raise TypeError('`{}` and `{}` are not of the same profiler type.'.
+                            format(type(self).__name__, type(other).__name__))
+
+        merged_profile = UnstructuredProfiler(
+            data=None, samples_per_update=self._samples_per_update,
+            min_true_samples=self._min_true_samples, options=self.options
+        )
+        merged_profile.encoding = self.encoding \
+            if self.encoding == other.encoding else 'multiple files'
+        merged_profile.file_type = self.file_type \
+            if self.file_type == other.file_type else 'multiple files'
+        merged_profile._empty_line_count = (
+            self._empty_line_count + other._empty_line_count)
+        merged_profile.total_samples = self.total_samples + other.total_samples
+
+        samples = list(dict.fromkeys(self.sample + other.sample))
+        merged_profile.sample = random.sample(list(samples),
+                                              min(len(samples), 5))
+        merged_profile._profile = self._profile + other._profile
+        return merged_profile
+
+    def _get_sample_size(self, data):
+        """
+        Determines the minimum sampling size for detecting column type.
+
+        :param data: data to be profiled
+        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame]
+        :return: integer sampling size
+        :rtype: int
+        """
+        if self._samples_per_update:
+            return self._samples_per_update
+
+        len_data = len(data)
+        if len_data <= self._min_sample_size:
+            return int(len_data)
+        return max(int(self._sampling_ratio * len_data), self._min_sample_size)
+
+    def _update_base_stats(self, base_stats):
+        """
+        Updates the samples and line count of the class for the given dataset
+        batch.
+
+        :param base_stats: dictionary of basic sampling / data stats
+        :type base_stats: dict
+        :return: None
+        """
+        self.total_samples += base_stats["sample_size"]
+        self.sample = base_stats["sample"]
```

Comment:
I understand this is how sample is currently saved in `Profiler` as well, but if this is called at the end of `_update_profile_from_chunk`, which comes after `update_profile`, then that means the sample always comes from the newest data profiled, should it do a random sampling of the 2 samples concatenated, like how it does in `__add__`?

Reply:
TBH i'm not sure, it kinda makes sense that it would only be from newest profile since that was the batch just processed via update, but could go either way. I think after these are merged we can make a decision which we apply to both cases.

Reply:
Fair enough, it just seems weird because imagine the data changes over time with updates, the sample would only reflect the latest data, which may not be a good representation of the whole set.

Reply:
agree with Chris. Should do sampling on the merged samples.

Reply:
If we decide to do this, should be in subsequent PR such that both are fixed together, there is more than just this that only looks at the recent for updates, e.g. file type etc

Reply:
yeah, another PR to do that is fine.